### ISSUE TITLE
`reportIncompatibleMethodOverride` false positive when one method is a `staticmethod` and the other is a `classmethod`

### ISSUE THREAD

**Issue by DetachHead** [DetachHead/basedpyright#1202]
### Description

Code sample in [basedpyright playground](https://basedpyright.com/?typeCheckingMode=all&code=MYGwhgzhAEBiD28BcBYAUNT0ACEAuYeAlsALYCmeAFvACbpbS3kBm0LiAFAJTQC0APmhEAdniTQAdNPTpQkGACEwAJ04J43VBizZ5UCtToMszNh3idQEXoOFiJ0yUA)

```python
class Foo:
    @staticmethod
    def foo() -> int: ...

class Bar(Foo):
    @classmethod
    def foo(cls) -> int: ...
```
upstream issue: https://github.com/microsoft/pyright/issues/8240

i don't get the reasoning, you can argue they are "technically" different but the interface for them is exactly the same, so i don't see how it could be in any way unsafe to allow this.
